src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/FinremCaseData.java

```diff
@@ -724,6 +724,18 @@ public boolean isAppAddressConfidential() {
         return YesOrNo.YES.equals(getContactDetailsWrapper().getApplicantAddressHiddenFromRespondent());
     }
 
+    @JsonIgnore
+    public String getApplicantSolicitorPostcode() {
+        return isConsentedApplication()
+            ? getContactDetailsWrapper().getSolicitorAddress().getPostCode()
+            : getContactDetailsWrapper().getApplicantSolicitorAddress().getPostCode();
+    }
+
+    @JsonIgnore
+    public String getRespondentSolicitorPostcode() {
+        return getContactDetailsWrapper().getRespondentSolicitorAddress().getPostCode();
+    }
```

Comment:
Looks like potentially NPE when calling `getRespondentSolicitorAddress`

Reply:
I'm not sure if it would be as we would be checking for null value on getContactDetailsWrapper?

Reply:
As above. The null value will come from getContactDetailsWrapper().getRespondentSolicitorAddress() rather than just getContactDetailsWrapper()

Reply:
[c76ba92](https://github.com/hmcts/finrem-case-orchestration-service/pull/1753/commits/c76ba927215850187445182fcc43f4d9bcef4345)